server/src/main/java/org/opensearch/action/search/DfsQueryPhase.java

```diff
@@ -69,7 +69,7 @@ final class DfsQueryPhase extends SearchPhase {
         Function<ArraySearchPhaseResults<SearchPhaseResult>, SearchPhase> nextPhaseFactory,
         SearchPhaseContext context
     ) {
-        super("dfs_query");
+        super(SearchPhaseName.DFS_QUERY.getName());
```

Comment:
Since we have introduced the `enum` for different, can we keep `enum` instead of `string` in the base class and use that everywhere (atleast in new code path)

Reply:
I tried exactly that. There were so many places where the code needs be changed, that it was going out of scope for this PR. Hence I thought not to do this for this PR, to keep changes under control.

The changes will be needed both in sec and tests files.